scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1985,7 +1985,8 @@ class FormatOps(
 
     trait Impl {
       def tryGetSplits(ft: FormatToken, nft: FormatToken)(implicit
-          style: ScalafmtConfig
+          style: ScalafmtConfig,
+          line: sourcecode.Line
```

Comment:
why do you need to pass the line? and if you really do, can that be a separate commit, with item description?

Reply:
I spend a bunch of time trying to figure out why we were adding a newline in line 2084 of Router and it turned out it wasn't in in at all. Maybe we could have the while line plus source file? I can move it to a separate commit.

Reply:
yeah, i know, I'm used to it by now. your idea of passing the filename is interesting, though, perhaps we should do that later.

Reply:
I wonder how to do it and not have to change every single:
```
implicit  line: sourcecode.Line
```

Into
```
implicit  line: sourcecode.Line,
implicit  file: sourcecode.File
```

Reply:
I removed all the added `line`

Reply:
https://github.com/scalameta/scalafmt/pull/2514 <- quick search and replace